src/ai/backend/client/cli/image.py

```diff
@@ -92,3 +92,40 @@ def forget(reference_or_id, arch):
             print_done(f"Image forgotten: {reference_or_id}")
         else:
             print_fail("Image forget has failed: {0}".format(result["msg"]))
+
+
+@image.command()
+@click.argument("reference_or_id", type=str)
+@click.option("--arch", type=str, default=None, help="Set an explicit architecture.")
+@click.option(
+    "--remove-from-registry",
+    is_flag=True,
+    default=False,
+    help="If set, removes the image from the registry. Only supported in Harbor registry.",
+)
+def purge(reference_or_id, arch, remove_from_registry):
+    """Delete image from server. This command will only work for image customized by user
+    unless callee has superadmin privileges.
+
+    REFERENCE_OR_ID: Canonical string of image (<registry>/<project>/<name>:<tag>)"""
+    with Session() as session:
+        try:
+            image_id = get_image_id(session, reference_or_id, architecture=arch)
+        except BackendAPIError:
+            print_fail("Could not find image.")
+            if not arch:
+                print_warn(
+                    "`arch` parameter not passed. If you are trying to resolve image via its canonical string you have to specify which architecture are you trying to manipulate with."
+                )
+            sys.exit(ExitCode.FAILURE)
+        try:
+            if remove_from_registry:
+                result = session.Image.untag_image_from_registry(image_id)
+                if not result["ok"]:
+                    print_fail("Image untag has failed: {0}".format(result["msg"]))
+                    sys.exit(ExitCode.FAILURE)
+            session.Image.purge_image_by_id(image_id)
```

Comment:
I guess it should be managed as a transaction, but is it okay if only one of the actions (remove_from_registry, purge_image_by_id) executes and the other fails?

Reply:
If `untag_image_from_registry` succeeds but `purge_image_by_id` fails, an inappropriate ImageRow will remain in the `images` table.

Unless the `purge_image` mutation is explicitly called for the corresponding `ImageRow`, it will continue to exist.

Ideally, a soft deletion process should be added during the image rescan to handle images that have already been removed from the registry.

See BA-969 for this